pg_search/src/postgres/options.rs

```diff
@@ -232,6 +268,33 @@ impl SearchIndexCreateOptions {
         ptr.as_ref().unwrap()
     }
 
+    /// Returns the configured `layer_sizes`, split into a [`Vec<u64>`] of byte sizes.
+    ///
+    /// If none is applied to the index, the specified `default` sizes are used.
+    pub fn layer_sizes(&self, default: &[u64]) -> Vec<u64> {
+        let layer_sizes_str = self.get_str(self.layer_sizes_offset, Default::default());
+        if layer_sizes_str.trim().is_empty() {
+            return default.to_vec();
+        }
+
+        let mut layer_sizes = Vec::new();
+        for part in layer_sizes_str.split(",") {
+            // just make sure postgres can parse this byte size
+            unsafe {
+                let byte_size =
+                    direct_function_call::<i64>(pg_sys::pg_size_bytes, &[part.into_datum()])
+                        .expect("`pg_size_bytes()` should not return NULL");
+
+                layer_sizes.push(
+                    byte_size
+                        .try_into()
+                        .expect("the layer size must fit in an `u64`"),
```

Comment:
The same thing, it's not about fitting `u64` as it has exactly the same size as `i64`, but the allowed range. It must be a positive value.